app/pvtable/src/main/java/org/phoebus/applications/pvtable/ui/PVTable.java

```diff
@@ -459,10 +459,12 @@ public void tableItemChanged(final PVTableItem item)
             final int row = model.getItems().indexOf(item);
 
             // System.out.println(item + " changed in row " + row + " on " + Thread.currentThread().getName());
-            final TableItemProxy proxy = rows.get(row);
-            if (proxy.getItem() != item)
-                throw new IllegalStateException("*** Looking for " + item.getName() + " but found " + proxy.name.get());
-            proxy.update(item);
+            if(row > -1  && row  < rows.size()) {
```

Comment:
You do add this safeguard in the PVTable code. Looks fine, but not sure what prompted this. Did you experience an index-out-of-bounds error here?